buildshiprun/handler.go

```diff
@@ -377,27 +387,28 @@ func deployFunction(deploy deployment, gatewayURL string, c *http.Client) (strin
 	return string(buildStatus), err
 }
 
+// enableStatusReporting check if status reportng is enabled
 func enableStatusReporting() bool {
 	return os.Getenv("report_status") == "true"
 }
 
-func reportStatus(status *sdk.Status) {
-	if !enableStatusReporting() {
-		return
-	}
-
-	gatewayURL := os.Getenv("gateway_url")
-
-	payloadSecret, keyErr := sdk.ReadSecret("payload-secret")
-	if keyErr != nil {
-		log.Printf("failed to load hmac key for status, error " + keyErr.Error())
-		return
-	}
+// initStatusReporter initialize status reporter
+func initStatusReporter() sdk.StatusReporter {
+	var reporter sdk.StatusReporter
+	if enableStatusReporting() {
+		hmacKey, keyErr := secret.Read("payload-secret")
+		if keyErr != nil {
+			log.Printf("failed to load hmac key for status, error " + keyErr.Error())
+			reporter = sdk.NilStatusReporter{}
+			return reporter
+		}
 
-	_, reportErr := status.Report(gatewayURL, payloadSecret)
-	if reportErr != nil {
-		log.Printf("failed to report status, error: %s", reportErr.Error())
+		gatewayURL := os.Getenv("gateway_url")
+		reporter = sdk.GithubStatusReporter{HmacKey: hmacKey, Gateway: gatewayURL}
+	} else {
+		reporter = sdk.NilStatusReporter{}
 	}
+	return reporter
 }
```

Comment:
Maybe we could somehow reuse the code from `github-push` or vice versa. What do you think @s8sg?

Reply:
Apologies for not selecting all lines which this comment refer too